internal/handlers/sqlite/sqlite.go

```diff
@@ -30,6 +32,14 @@ import (
 	"github.com/FerretDB/FerretDB/internal/util/state"
 )
 
+var (
```

Comment:
Why do we want to differentiate those two errors?

Reply:
That's how MongoDB works. They return different error message for those cases.

Reply:
Well, I believe MongoDB doesn't return "invalid FerretDB collection name" error message :) For simplicity, I'd suggest we just return "invalid FerretDB collection name"  in both cases and not differentiate it. We use the same `ErrCollectionStartsWithDot` error code in both error messages anyway. Otherwise, we might want to have a separate error message for a case when both MongoDB and FerretDB consider the collection name too long etc. For simplicity, for now, I'd just keep the same error message not to introduce unclarity or technical debt. Otherwise, if you want proper error messages, please create a separate issue and describe there how they should work.

Reply:
To simplify error handling and stay consistent with MongoDB handling I've used backends.Error as sugested [here](https://github.com/FerretDB/FerretDB/pull/2868#discussion_r1243560985). WDYT?

Reply:
Yeah, sounds good to me!